src/julia-syntax.scm

```diff
@@ -2422,13 +2422,18 @@
 
    'opaque_closure
    (lambda (e)
-     (let* ((ty   (and (length> e 2) (expand-forms (cadr e))))
+     (let* ((ty   (and (length> e 2) (if (eq? (caadr e) '->) (cadr e)
+                    (error "Opaque closure type must be specified in the form Tuple{T,U...}->RT"))))
             (F    (if (length> e 2) (caddr e) (cadr e)))
+            (argt (let ((a (and ty (expand-forms (cadr ty)))))
+                       (and (not (eq? a '_)) a)))
+            (rett (let ((body (and ty (expand-forms (caddr (caddr ty))))))
```

Comment:
I'm somewhat blinding stepping over the `linenum` in the block with the `caddr` here

So this assumes that our `->` lowering always lowers to a block with exactly one leading linenum

Reply:
OK, that sounds like a good idea to me

Maybe we should only support the fully-expanded intermediate form, as `Expr(:opaque_closure, AT, RT_LB, RT_UB, ex)`?